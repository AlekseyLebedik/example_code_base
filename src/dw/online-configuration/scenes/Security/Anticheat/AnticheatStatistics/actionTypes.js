export const ANTICHEAT_STATISTICS_CHANGE_USER_ID =
  'security/ANTICHEAT_STATISTICS_CHANGE_USER_ID';
export const ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID =
  'security/ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID';
export const ANTICHEAT_STATISTICS_CHANGE_DATE =
  'security/ANTICHEAT_STATISTICS_CHANGE_DATE';
