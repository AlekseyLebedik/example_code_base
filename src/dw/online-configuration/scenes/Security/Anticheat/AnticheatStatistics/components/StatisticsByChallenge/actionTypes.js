export const ANTICHEAT_STATISTICS_FETCH =
  'security/ANTICHEAT_STATISTICS_BY_CHALLENGE_FETCH';
export const ANTICHEAT_STATISTICS_FETCH_SUCCESS =
  'security/ANTICHEAT_STATISTICS_BY_CHALLENGE_FETCH_SUCCESS';
