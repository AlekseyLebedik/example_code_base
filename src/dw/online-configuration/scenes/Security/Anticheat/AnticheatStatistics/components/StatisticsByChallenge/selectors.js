export const anticheatStatisticsSelector = state =>
  state.Scenes.Security.Anticheat.AnticheatStatistics.ByChallenge
    .anticheatStatistics;
