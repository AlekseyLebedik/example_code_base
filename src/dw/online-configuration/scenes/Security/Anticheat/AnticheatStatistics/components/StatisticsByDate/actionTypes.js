export const ANTICHEAT_STATISTICS_FETCH =
  'security/ANTICHEAT_STATISTICS_BY_DATE_FETCH';
export const ANTICHEAT_STATISTICS_FETCH_SUCCESS =
  'security/ANTICHEAT_STATISTICS_BY_DATE_FETCH_SUCCESS';
