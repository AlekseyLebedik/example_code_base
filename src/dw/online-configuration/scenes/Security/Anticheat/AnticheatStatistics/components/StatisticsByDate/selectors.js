import { createSelector } from 'reselect';
import { DATE_TIME_FORMATS, formatDateTime } from 'dw/core/helpers/date-time';

const statisticsSelector = state =>
  state.Scenes.Security.Anticheat.AnticheatStatistics.ByDate
    .anticheatStatistics;

export const anticheatStatisticsSelector = createSelector(
  statisticsSelector,
  anticheatStatistics =>
    anticheatStatistics.map(anticheatStatistic => ({
      ...anticheatStatistic,
      date: formatDateTime(
        anticheatStatistic.date,
        DATE_TIME_FORMATS.DEFAULT_DATE
      ),
    }))
);
