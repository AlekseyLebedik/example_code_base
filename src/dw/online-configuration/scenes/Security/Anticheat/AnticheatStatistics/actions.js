import * as AT from './actionTypes';

export const changeUserId = userID => ({
  type: AT.ANTICHEAT_STATISTICS_CHANGE_USER_ID,
  userID,
});

export const changeChallengeId = challengeId => ({
  type: AT.ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID,
  challengeId,
});

export const changeDate = date => ({
  type: AT.ANTICHEAT_STATISTICS_CHANGE_DATE,
  date,
});
