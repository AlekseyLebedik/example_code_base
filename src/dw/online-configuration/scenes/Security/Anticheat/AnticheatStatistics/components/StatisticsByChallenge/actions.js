import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import * as AT from './actionTypes';

export const fetchAnticheatStatistics = (params, append = false) => ({
  type: AT.ANTICHEAT_STATISTICS_FETCH,
  params,
  append,
});

export const fetchAnticheatStatisticsSuccess = (payload, append) => ({
  type: AT.ANTICHEAT_STATISTICS_FETCH_SUCCESS,
  payload,
  append,
});

export const fetchAnticheatStatisticsFailed = err => dispatch =>
  dispatch(nonCriticalHTTPError(err));
