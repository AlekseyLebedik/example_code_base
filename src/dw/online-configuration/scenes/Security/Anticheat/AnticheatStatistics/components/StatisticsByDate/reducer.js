import * as AT from './actionTypes';

const INITIAL_STATE = {
  anticheatStatistics: [],
  nextPageToken: null,
};

const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.ANTICHEAT_STATISTICS_FETCH_SUCCESS:
      return {
        ...state,
        anticheatStatistics: action.append
          ? [...state.anticheatStatistics, ...action.payload.data]
          : action.payload.data,
        nextPageToken: action.payload.nextPageToken || null,
      };
    default:
      return state;
  }
};

export default reducer;
