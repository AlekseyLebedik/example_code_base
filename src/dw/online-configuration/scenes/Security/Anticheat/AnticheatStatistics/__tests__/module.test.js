import * as actions from '../actions';
import * as AT from '../actionTypes';
import * as selectors from '../selectors';

const userId = '1234567890';
const challengeId = 1;
const date = new Date('2018-07-12T01:01:01.000Z');

describe('Security - Anticheat - AnticheatStatistics', () => {
  describe('Action Creators', () => {
    it('ANTICHEAT_STATISTICS_CHANGE_USER_ID', () => {
      const action = actions.changeUserId(userId);

      expect(action).toHaveProperty(
        'type',
        AT.ANTICHEAT_STATISTICS_CHANGE_USER_ID
      );
      expect(action).toHaveProperty('userID', userId);
    });

    it('ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID', () => {
      const action = actions.changeChallengeId(challengeId);

      expect(action).toHaveProperty(
        'type',
        AT.ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID
      );
      expect(action).toHaveProperty('challengeId', challengeId);
    });

    it('ANTICHEAT_STATISTICS_CHANGE_DATE', () => {
      const action = actions.changeDate(date);

      expect(action).toHaveProperty(
        'type',
        AT.ANTICHEAT_STATISTICS_CHANGE_DATE
      );
      expect(action).toHaveProperty('date', date);
    });
  });

  describe('Reducer', () => {
    let _reducer = null;
    const fixedDate = new Date('2018-02-28T00:00:00.000Z');

    beforeAll(() => {
      import('../reducer').then(anticheatStatisticsReducer => {
        _reducer = anticheatStatisticsReducer.default;
      });
      global.Date = class extends global.Date {
        constructor() {
          super();
          return fixedDate;
        }
      };
    });

    it('return the initial state', () => {
      expect(_reducer(undefined, {})).toMatchSnapshot();
    });

    it('handle ANTICHEAT_STATISTICS_CHANGE_USER_ID', () => {
      expect(
        _reducer(undefined, actions.changeUserId(userId))
      ).toMatchSnapshot();
    });

    it('handle ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID', () => {
      expect(
        _reducer(undefined, actions.changeChallengeId(challengeId))
      ).toMatchSnapshot();
    });

    it('handle ANTICHEAT_STATISTICS_CHANGE_DATE', () => {
      expect(_reducer(undefined, actions.changeDate(date))).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    it('validate userIdSelector', () => {
      const state = {
        Scenes: {
          Security: {
            Anticheat: { AnticheatStatistics: { Filters: { userID: userId } } },
          },
        },
      };
      expect(selectors.userIdSelector(state)).toBe(userId);
    });

    it('validate challengeIdSelector', () => {
      const state = {
        Scenes: {
          Security: {
            Anticheat: { AnticheatStatistics: { Filters: { challengeId } } },
          },
        },
      };
      expect(selectors.challengeIdSelector(state)).toBe(challengeId);
    });

    it('validate dateSelector', () => {
      const state = {
        Scenes: {
          Security: {
            Anticheat: { AnticheatStatistics: { Filters: { date } } },
          },
        },
      };
      expect(selectors.dateSelector(state)).toBe(date);
    });
  });
});
