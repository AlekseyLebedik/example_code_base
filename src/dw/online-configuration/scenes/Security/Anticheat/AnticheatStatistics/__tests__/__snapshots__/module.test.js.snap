// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security - Anticheat - AnticheatStatistics Reducer handle ANTICHEAT_STATISTICS_CHANGE_CHALLENGE_ID 1`] = `
Object {
  "ByChallenge": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByDate": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByUser": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "Filters": Object {
    "challengeId": 1,
    "date": 1519776000,
    "userID": null,
  },
}
`;

exports[`Security - Anticheat - AnticheatStatistics Reducer handle ANTICHEAT_STATISTICS_CHANGE_DATE 1`] = `
Object {
  "ByChallenge": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByDate": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByUser": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "Filters": Object {
    "challengeId": null,
    "date": 2018-07-12T01:01:01.000Z,
    "userID": null,
  },
}
`;

exports[`Security - Anticheat - AnticheatStatistics Reducer handle ANTICHEAT_STATISTICS_CHANGE_USER_ID 1`] = `
Object {
  "ByChallenge": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByDate": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByUser": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "Filters": Object {
    "challengeId": null,
    "date": 1519776000,
    "userID": "1234567890",
  },
}
`;

exports[`Security - Anticheat - AnticheatStatistics Reducer return the initial state 1`] = `
Object {
  "ByChallenge": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByDate": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "ByUser": Object {
    "anticheatStatistics": Array [],
    "nextPageToken": null,
  },
  "Filters": Object {
    "challengeId": null,
    "date": 1519776000,
    "userID": null,
  },
}
`;
